msmtools/util/sorted_schur.py

```diff
@@ -23,111 +23,64 @@ def _initialize_matrix(M, P):
-#         if np.any(np.isnan(eigenvals)):
-#             raise ValueError("Some eigenvalues of P are NaN.")
-#         if (z == 'LM'):
-#             idx = np.argsort(np.abs(eigenvals))
-#             sorted_eigenvals = eigenvals[idx]
-#             top_eigenvals = sorted_eigenvals[::-1][:m+1]
-#         elif (z == 'LR'):
-#             idx = np.argsort(np.real(eigenvals))
-#             sorted_eigenvals = eigenvals[idx]
-#             top_eigenvals = sorted_eigenvals[::-1][:m+1]
-#     else: 
-    from petsc4py import PETSc
-    from slepc4py import SLEPc 
-    
-    # Initialize boolean to indicate, if a 2x2-block is split.
-    block_split = False
-    
-    M = PETSc.Mat().create()
-    _initialize_matrix(M, P)
-    # Creates EPS object.
-    E = SLEPc.EPS()
-    E.create()
-    # Set the matrix associated with the eigenvalue problem.
-    E.setOperators(M)
-    # Select the particular solver to be used in the EPS object: Krylov-Schur
-    E.setType(SLEPc.EPS.Type.KRYLOVSCHUR)
-    # Set the number of eigenvalues to compute and the dimension of the subspace.
-    E.setDimensions(nev=k)
-    # set the tolerance used in the convergence criterion
-    E.setTolerances(tol=tol)
-    if z == 'LM':
-        E.setWhichEigenpairs(E.Which.LARGEST_MAGNITUDE)
-    elif z == 'LR':
-        E.setWhichEigenpairs(E.Which.LARGEST_REAL)
-    else:
-        raise ValueError(f"Invalid spectrum sorting options `{z}`. Valid options are: `'LM'`, `'LR'`")
-    # Solve the eigensystem.
-    E.solve()
+    return np.isclose(eigenval_in, np.conj(eigenval_out))
 
-    # Gets the number of converged eigenpairs. 
-    nconv = E.getConverged()
-    # Warn, if nconv smaller than m.
-    if nconv < k:
-        warnings.warn(f"The number of converged eigenpairs `nconv={nconv}` is too small.")
-    # Collect the m dominant eigenvalues.
-    top_eigenvals = []
-    top_eigenvals_error = []
-    for i in range(nconv):
-        # Get the i-th eigenvalue as computed by solve().
-        eigenval = E.getEigenvalue(i)
-        top_eigenvals.append(eigenval)
-        # Computes the error (based on the residual norm) associated with the i-th computed eigenpair.
-        eigenval_error = E.computeError(i)
-        top_eigenvals_error.append(eigenval_error)
-    top_eigenvals = np.asarray(top_eigenvals)
-    top_eigenvals_error = np.asarray(top_eigenvals_error)
-    
-    if (m < n):
-        eigenval_in = top_eigenvals[m-1]
-        eigenval_out = top_eigenvals[m]
-        # Don't separate conjugate eigenvalues (corresponding to 2x2-block in R).
-        if np.isclose(eigenval_in, np.conj(eigenval_out)):
-            block_split = True
-            warnings.warn("Clustering into " + str(m) + " clusters will split conjugate eigenvalues! "
-                          "Request one cluster more or less.")
-                
-    return top_eigenvals, block_split
-
-
-def sorted_krylov_schur(P, m, z='LM', tol=1e-16):
+
+def _check_schur(P, Q, R, eigenvalues, method = ""):
```

Comment:
Why do you use `method = ""` instead of `method`? A method should be passed in any case, right?

Reply:
Since the `method` parameter is only used for printing in the `_check_schur` function, I made it optional so that I could use the function more flexibly, e.g. when I want to call it from outside `sorted_schur.py`.

Reply:
I don't have a strong opinion here, if you want, I can also make it mandatory to pass the method string.

Reply:
I guess it doesn't matter from point of safety. So do as you like :)

Reply:
alright, I left it the way it had done it originally.